fastly/block_fastly_waf_configuration_v1_exclusion.go

```diff
@@ -0,0 +1,212 @@
+package fastly
+
+import (
+	"fmt"
+	gofastly "github.com/fastly/go-fastly/v2/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+	"strconv"
+)
+
+var wafRuleExclusion = &schema.Schema{
+	Type:     schema.TypeSet,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the exclusion.",
+			},
+			"condition": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "A conditional expression in VCL used to determine if the condition is met.",
+			},
+			"exclusion_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The type of exclusion.",
+				ValidateFunc: validateExecutionType(),
+			},
+			"modsec_rule_ids": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "The modsec rule ids to exclude.",
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+			"number": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "A sequential ID assigned to the exclusion.",
+			},
+		},
+	},
+}
+
+func readWAFRuleExclusions(meta interface{}, d *schema.ResourceData, wafVersionNumber int) error {
+	conn := meta.(*FastlyClient).conn
+	wafID := d.Get("waf_id").(string)
+
+	resp, e := conn.ListAllWAFRuleExclusions(&gofastly.ListAllWAFRuleExclusionsInput{
+		WAFID:            wafID,
+		WAFVersionNumber: wafVersionNumber,
+		Include:          strToPtr("waf_rules"),
```

Comment:
This may need to change to a slice, pending discussion on the go-fastly PR.

Reply:
Updated